File: go/dir/credentials.go
package dir

import "crypto/tls"

// Credentials abstracts the TLS posture used by the DIR client.
//
// Two stock impls (InsecureCredentials, TlsCredentials) ship in this
// package. SPIFFE-backed credentials live in hop.top/agntcy/spiffe.
type Credentials interface {
	TLS() *tls.Config
}

// InsecureCredentials disables TLS. Dev/test only.
type InsecureCredentials struct{}

func (InsecureCredentials) TLS() *tls.Config { return nil }

// TlsCredentials carries a stdlib *tls.Config.
type TlsCredentials struct {
	Config *tls.Config
}

func (c TlsCredentials) TLS() *tls.Config { return c.Config }

File: sdk/rs/poly-agntcy-dir/src/credentials.rs
use std::sync::Arc;

/// TLS posture for [`crate::client::Client`].
pub trait Credentials: Send + Sync {
    fn tls(&self) -> Option<Arc<rustls::ClientConfig>>;
}

/// No TLS. Dev/test only.
pub struct InsecureCredentials;

impl Credentials for InsecureCredentials {
    fn tls(&self) -> Option<Arc<rustls::ClientConfig>> {
        None
    }
}

/// Wraps a pre-built rustls ClientConfig.
pub struct TlsCredentials {
    pub config: Arc<rustls::ClientConfig>,
}

impl Credentials for TlsCredentials {
    fn tls(&self) -> Option<Arc<rustls::ClientConfig>> {
        Some(self.config.clone())
    }
}
